http/digest/src/test/java/org/wildfly/security/http/digest/DigestMechanismFactoryTest.java

```diff
@@ -0,0 +1,71 @@
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.wildfly.security.http.HttpAuthenticationException;
+import org.wildfly.security.http.HttpServerAuthenticationMechanism;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import java.io.IOException;
+import java.util.HashMap;
+
+public class DigestMechanismFactoryTest {
+
+    private DigestMechanismFactory digestMechanismFactory = new DigestMechanismFactory();
+
+    CallbackHandler dummyCallbackHandler = new CallbackHandler() {
+        @Override
+        public void handle(Callback[] callbacks) throws IOException, UnsupportedCallbackException {
+        }
+    };
+
+    @Test
+    public void testGetMechanismNamesPropertiesNull(){
+        digestMechanismFactory.getMechanismNames(null);
+        digestMechanismFactory.getMechanismNames(new HashMap<String,String>());
+    }
+
+    @Test
+    public void testGetMechanismNamesReturnNotNull(){
+        String[] mechanismNames = digestMechanismFactory.getMechanismNames(null);
+        Assert.assertNotNull("Array of mechanism names is not null.", mechanismNames);
+    }
+
+    @Test
+    public void testCreateAuthenticationMechanismMechanismNameNull() throws HttpAuthenticationException{
+        try {
+            digestMechanismFactory.createAuthenticationMechanism(null,new HashMap<String,String>(),dummyCallbackHandler);
+            Assert.fail("Mechanism name could not be null");
+        }catch (IllegalArgumentException illegalArgumentException){
+            // OK - expected exception state
+        }
+    }
+
+    @Test
+    public void testCreateAuthenticationMechanismPropertiesNull() throws HttpAuthenticationException{
+        try {
+            digestMechanismFactory.createAuthenticationMechanism("DIGEST",null,dummyCallbackHandler);
+            Assert.fail("Properties could not be null");
+        }catch (IllegalArgumentException illegalArgumentException){
+            // OK - expected exception state
+        }
+    }
+
+    @Test
+    public void testCreateAuthenticationMechanismCallbackHandlerNull() throws HttpAuthenticationException{
+        try {
+            digestMechanismFactory.createAuthenticationMechanism("DIGEST",new HashMap<String,String>(),null);
+            Assert.fail("CallbackHandler could not be null");
+        }catch (IllegalArgumentException illegalArgumentException){
+            // OK - expected exception state
+        }
+    }
+
+    @Test
+    public void testCreateAuthenticationMechanismWrongMechanismName() throws HttpAuthenticationException{
+        HttpServerAuthenticationMechanism httpServerAuthenticationMechanism = digestMechanismFactory.createAuthenticationMechanism("MECHANISM_NAME_DOES_NOT_EXISTS",new HashMap<String,String>(),dummyCallbackHandler);
+        Assert.assertNull("Provided mechanism must be null.", httpServerAuthenticationMechanism);
+    }
+}
```

Comment:
Hi @keshav-redhat , the PR looks good! Just a minor, there is this task in the issue description that is still missing:

```
Also add a test case to both which tests that if different challenge prefix gets used (so eg. try to use BasicMechanismFactory to create authentication mechanism with name "DIGEST" and vice versa), the factory returns null. But with appropriate prefix the factory will create digest / basic authentication mechanism as expected.
```

What I meant by that is, that it would be nice to add a test here that uses prefix `BASIC` and make sure the `DigestMechanismFactory` returns null in such a case as it dos not provide such a mechanism. 

Another test that could be added here is the case where we have a correct mechanism name, non-empty properties and existing callback and make sure that the resulting HttpServerAuthenticationMechanism is not null.

Reply:
Hello @keshav-redhat  , this was not updated